### ISSUE TITLE
Make empty list comparisons more Pythonic and faster.

### ISSUE THREAD

**Issue by krnick** [quark-engine/quark-engine#53]
There are three methods to check whether the list is empty in Python.

1. its Length

```
if len(a)==0:
    print("Empty")
```

2. Direct Comparison

```
if my_list == []:
    print("Empty")
```

3. Type Flexibility

```
if not a_list:
    print("Empty")
```

### Reference:

Empty lists evaluate to False ([Link](https://www.tutorialspoint.com/How-to-check-if-a-list-is-empty-in-Python))

Performance evaluation [test](https://therenegadecoder.com/code/how-to-check-if-a-list-is-empty-in-python/#performance)

[Writing faster Python](https://youtu.be/YjHsOrOOSuI?t=1045) in EuroPython Conference

### In conclusion:

Type Flexibility is faster and more Pythonic than the other two methods.

* Not Empty

```
if a_list:
    print("Not Empty")
```

* Empty
```
if not a_list:
    print("Empty")
```
